### PR TITLE
Update api.go

### LINKED ISSUE #38: AWS Translate code not working: zh-TW, fa-AF, fr-CA, es-MX
For the language codes that has a hyphen in between, /autotranslate target [code] will respond with no error, but an attempt to translate will result in an error: Invalid parameter: target  (close)

these languages are essentially not working

### PR DESCRIPTION
**Pull request by daniron26**
#### Summary
Change language code error test from len==2 to 2 <= len <= 5

the current version will reject correct language codes like "fr-CA" and other language codes that has length == 5 

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-plugin-autotranslate/issues/38

### CODE REVIEW

**Comment by saturninoabril** [comment]
Awesome, thanks @daniron26! Will take a look at this.

File: server/api.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func (p *Plugin) getGo(w http.ResponseWriter, r *http.Request) {
 	}
 
 	target := r.URL.Query().Get("target")
-	if len(target) != 2 {
+	if len(target) < 2 || len(target) > 5 {
```
Comment (hanzei) [inline server/api.go:69]: Are the code that have the length 3 or 4?

**Review by hanzei — CHANGES_REQUESTED** [review]
Just one question

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @daniron26! LGTM. Could you please do the same for [source](https://github.com/mattermost/mattermost-plugin-autotranslate/pull/40/files#diff-3fb3d4184590bc6f747661f3062676031e837b0dd5f2b18640f439be7a2235f6L63)?

File: server/api.go (hunk 1 above)
Comment (saturninoabril) [reply server/api.go:69]: Looking at some typical reference online and currently [supported language by AWS Translate](https://docs.aws.amazon.com/translate/latest/dg/what-is.html#language-pairs), it's either:
- 2 (ex. `fr`), or
- 5 (ex. `fr-CA`), including dash

Comment (daniron26) [reply server/api.go:69]: I don't think there are len 3 or 4 codes, only 2 and 5

**Review by daniron26 — COMMENTED** [review]
source language code now allows for 5 character codes like fr-CA

**Review by daniron26 — COMMENTED** [review]
CI failed, not sure how to proceed

File: server/api.go [hunk 2]
```diff
@@ -60,13 +60,13 @@ func (p *Plugin) getGo(w http.ResponseWriter, r *http.Request) {
 	}
 
 	source := r.URL.Query().Get("source")
-	if len(source) < 2 || len(source) > 4 {
+	if len(source) < 2 || len(source) > 5 {
```
Comment (hanzei) [inline server/api.go:63]: Should we just check for 
```suggestion
	if len(source) != 2 && len(source) != 5 {
```

here and below to be more strict?

**Review by hanzei — CHANGES_REQUESTED** [review]

File: server/api.go [hunk 3]
```diff
@@ -60,13 +60,13 @@ func (p *Plugin) getGo(w http.ResponseWriter, r *http.Request) {
...
 	source := r.URL.Query().Get("source")
-	if len(source) < 2 || len(source) > 4 {
+	if len(source) != 2 && len(source) != 5 {
 		http.Error(w, "Invalid parameter: source", http.StatusBadRequest)
 		return
 	}
 
 	target := r.URL.Query().Get("target")
-	if len(target) != 2 {
+	if len(target) < 2 || len(target) > 5 {
```
Comment (hanzei) [inline server/api.go:69]: Same here?

**Comment by saturninoabril** [comment]
@daniron26 Thanks for updating. Could you please confirm if the recent change works for you?

**Review by hanzei — APPROVED** [review]

**Comment by daniron26** [comment]
> Thanks @daniron26! LGTM. Could you please do the same for [source](https://github.com/mattermost/mattermost-plugin-autotranslate/pull/40/files#diff-3fb3d4184590bc6f747661f3062676031e837b0dd5f2b18640f439be7a2235f6L63)?

Its been changed, let me know if I missed anything.

> @daniron26 Thanks for updating. Could you please confirm if the recent change works for you?

Yes it does, happy to be a part of this.

**Comment by saturninoabril** [comment]
@daniron26 I'm getting the following error since the `source` could have a value of `auto`:
Screen Shot 2021-02-16 at 2 44 03 PM

With that, we can probably relax the length check in `source`.

**Comment by daniron26** [comment]
> @daniron26 I'm getting the following error since the `source` could have a value of `auto`:
> Screen Shot 2021-02-16 at 2 44 03 PM
> 
> With that, we can probably relax the length check in `source`.

I think so, changing it back to 2 >= len >= 5

File: server/api.go [hunk 4]
```diff
@@ -60,13 +60,13 @@ func (p *Plugin) getGo(w http.ResponseWriter, r *http.Request) {
 	}
 
 	source := r.URL.Query().Get("source")
-	if len(source) < 2 || len(source) > 4 {
+	if len(source) < 2 || len(source) > 5 {
```
Comment (saturninoabril) [inline server/api.go]: Failed on lint due to extra whitespace
```suggestion
	if len(source) < 2 || len(source) > 5 {
```

**Review by saturninoabril — APPROVED** [review]
Thanks @daniron26! Tested and passed. LGTM.

**Pull request merged**